src/app/features/courses/courses.component.ts

```diff
@@ -15,62 +16,73 @@ import { combineLatest, Subscription } from 'rxjs';
-  notFound = false;
+  isLoading$ = this.coursesStore.isLoading$ || this.coursesFacade.isAllCoursesLoading$;
+  searching = false;
+  filteredCourses: Course[] = [];
+  showInfo = '';
 
   constructor(
     private coursesStore: CoursesStoreService,
     private mapping: MappingService,
     private userStore: UserStoreService,
     private router: Router,
     private route: ActivatedRoute,
-    private authService: AuthService
+    private authService: AuthService,
+    private coursesFacade: CoursesStateFacade
   ) {}
 
   ngOnInit() {
     if (this.authService.isAuthorised) {
-      this.coursesStore.getAll();
+      this.coursesFacade.getAllCourses();
       this.coursesStore.getAllAuthors();
     } // else navigate to login but it is done by authGuard
-    this.subscription = combineLatest([this.coursesStore.courses$, this.coursesStore.authors$]).subscribe(
-      ([courses, authors]) => {
-        this.allCourses = courses;
-        this.allAuthors = authors;
+    this.subscription = combineLatest([
+      this.coursesFacade.allCourses$,
+      this.coursesFacade.courses$,
+      this.coursesFacade.isSearchingState$,
+      this.coursesStore.authors$,
+    ]).subscribe(([courses, filteredCourses, isSearching, authors]) => {
+      this.allCourses = courses;
+      this.allAuthors = authors;
+      this.searching = isSearching;
+      this.filteredCourses = filteredCourses;
+      if (this.searching && this.filteredCourses.length > 0) {
+        this.courses = this.mapping.createCoursesWithAuthorNames(this.filteredCourses, this.allAuthors);
+        this.showInfo = '';
+      } else if (this.searching && this.filteredCourses.length <= 0) {
         this.courses = this.mapping.createCoursesWithAuthorNames(this.allCourses, this.allAuthors);
+        this.showInfo = this.noCourseText;
+        this.triggerAnimation();
+      } else {
+        this.courses = this.mapping.createCoursesWithAuthorNames(this.allCourses, this.allAuthors);
+        this.showInfo = '';
       }
-    );
+    });
   }
 
+  readonly noCourseText = 'No Course found. All courses are shown';
   readonly emptyListTitle = 'Your List is Empty';
   readonly emptyListText = "Please use 'ADD NEW COURSE' button to add your first course";
 
-  handleShowCourseInfo(courseId: string) {
-    this.courseInfo = this.courses.find(course => course.id === courseId);
-  }
-  handleShowCourses(): void {
-    this.courseInfo = undefined;
-  }
-  handleSearch(searchText: string): void {
-    this.lastSearchedText = searchText;
-    const filterParams: FilterCourse =
-      searchText.length > 0
-        ? { title: [searchText], duration: [], description: [], creationDate: [] }
-        : { title: [], duration: [], description: [], creationDate: [] };
-    this.coursesStore.filterCourses(filterParams).subscribe(response => {
-      if (response.successful && response.result.length > 0) {
-        this.notFound = false;
-      } else this.notFound = true;
-    });
+  triggerAnimation() {
```

Comment:
This is tricky here. On first search click `Triggered undefined`appears on the console, on the subsequent calls the `Triggered ElementRef` is there. Moreover, the `element.classList` direct manipulation with `setTimeout` does the trick, but not ideal. I suggest you take a look at Angular Animation - which was not on the plate, yet.
Consider this comment resolved once you answer the question on why this is logging `undefined` on the first search click.
![image](https://github.com/user-attachments/assets/204ba475-2862-4963-9c0d-597ac64bfe8c)

Reply:
Nice to see the built in api for animation. Did not know. Thank you!
I made with that. 

For the question, I guess it should have been undefined because the trigger function is evaluated before the element would have been existing in the DOM . And trick was that it does not cause problem as for the first render we set animate class too... So at first render animation will run automatically.

Reply:
Yes, this looks so much better.
As long as the `*ngIf` conditional evaluates to false, the `span` element is not present on the DOM, therefore there is nothing to manipulate in your .ts.